pairlists.md

```diff
@@ -117,6 +117,33 @@ SEXP Rf_nthcdr(SEXP, int);  # access nth element
 
 If you use these it's easy to get O(n^2) behaviour, but for the usual size of pairlists, this is unlikely to cause a performance bottleneck.
 
+### Creating a call
+
+Pairlists are most commonly used for function calls. Calls to functions with 0 to 5 arguments can be created directly via `Rf_lang1()` to `Rf_lang6()`. If you have more than 5 arguments, or if you want to use named arguments, use the following template:
+```cpp
+// equivalent of:
+// fun(arg1 = a, arg2 = b, arg3 = c)
+PROTECT(call = Rf_allocVector(LANGSXP, 4)); // 4 = # of args + 1 
```

Comment:
I think keeping the assignment outside of PROTECT is more idiomatic and also works with inline type declaration:

```suggestion
SEXP call = PROTECT(Rf_allocVector(LANGSXP, 4)); // 4 = # of args + 1 
```

Reply:
Thanks, I'll make this change. I've seen both patterns in the R source code and was looking for a general principle for why to use one or the other. The one benefit I see for putting the `PROTECT()` around the assignment is that it moves all `PROTECT()` calls to the beginning of each line which makes it easy to count them. On the flip side, it may create the impression that `PROTECT()` somehow protects everything that is going on inside the parentheses, which is not true. (If there are nested calls, only the final pointer is protected.) Maybe some text on this should be added to the document covering the GC.